src/mock-doc/selector.ts

```diff
@@ -1,16 +1,72 @@
 import { MockElement } from './node';
 import jQuery from './third-party/jquery';
 
-export function matches(selector: string, elm: MockElement) {
-  const r = jQuery.find(selector, undefined, undefined, [elm]);
-  return r.length > 0;
+const JQUERY_SCOPE_ADVISORY = [
+  "At present jQuery does not support the ':scope' selector.",
+  'If you need to use it in your test, consider writing an end-to-end test instead.',
+].join('\n');
+
+/**
+ * Given a selector and an error object thrown by jQuery, annotate the
+ * error's message to add some context as to the probable reason for the error.
+ * In particular, if the selector includes the `:scope` selector, then we know
+ * that was likely the cause of the error.
+ *
+ * @param selector our selector of interest
+ * @param e an error object that was thrown in the course of using jQuery
+ */
+function updateScopeSelectorError(selector: string, e: unknown) {
+  if (selector.includes(':scope') && (e as Error).message) {
```

Comment:
I think there are a couple other L4 selectors that Sizzle doesn't handle on its own right now - `:where()` and `:is()`. What do you think about expanding this to query a list of known selectors?

I threw together a rough little patch since it'd take a little change on a couple of different lines/make it hard to look at in GH. TAL and let me know what you think!
[multi-l4-selector.patch](https://github.com/ionic-team/stencil/files/14131219/multi-l4-selector.patch)

Reply:
makes sense - just pushed that change